docs/cdoc/qk-classical-register.rst

```diff
@@ -0,0 +1,37 @@
+===================
+QkClassicalRegister
+===================
+
+.. code-block:: c
+
+   typedef struct QkClassicalRegister QkClassicalRegister
+
+A common way to instantiate several bits at once is to create a register, such
+as by:
+
+.. code-block: c
+
+    #include <qiskit.h>
+
+    QkClassicalRegister *creg = qk_classical_register_new(5, "my_creg");
+
+This has the advantage that you can give a collection of bits a name. This is
+used as metadata around bits in the circuit when you add the register to the
+circuit which will be used when exporting the circuit to interchange languages.
+Continuing from the above example:
```

Comment:
The wording of this is a little iffy, it can make a little more sense with a deep polishing, here's my quick attempt at getting there haha:

```suggestion
This has the advantage of giving a collection of bits a name, which is
used as metadata for the bits once the register is added to a circuit.
The assigned metadata will be used when exporting the circuit to
interchange languages.
Continuing from the above example:
```

This can be improved further